contracts/Contribution.sol

```diff
@@ -107,32 +110,26 @@ contract Contribution {
     }
 
     function _contribute(bool combine) internal {
-        require(isKeySet, "Key has not been set.");
+        require(isKeySet, "Material is not ready for contributions yet.");
         require(!materialReleaseConditionMet || block.timestamp < deadline,
             "Cannot contribute after the deadline");
         require(msg.value >= minContribution,
             "Contribution must be equal to or greater than the minimum.");
 
-        if (contributionsByAddress[msg.sender].length == 0) { // If this is the first contribution from this address
-            contributors.push(msg.sender); // Add the address to the contributors array
-        }
-
-        if (combine) {
-            contributionsByAddress[msg.sender][0] += msg.value;
-        } else {
-            contributionsByAddress[msg.sender].push(msg.value); // Add contribution to the mapping
-        }
+        contributionAmounts.push(msg.value);
+        contributorsForEachAContribution.push(msg.sender);
+        contributionIsCombined.push(combine);
```

Comment:
What is the meaning of the `combine` flag?

For example if I make the following contributions:
- contribute(0.5)
- contributeAndCombine(0.2)
- contributeAndCombine(0.4)
- contribute(0.1)
- contribute(0.25)
- contributeAndCombine(0.6)
- contribute(1)

What should my contributions be? Should it be:

### `combine` means "combine all my previous contributions into one"
Therefore I'd have 1 contribution with value `2.05` eth after the last `contributeAndCombine(0.6)` and one contribution with value `1` eth.


### `combine` flag set to true means combine with a first "default" contribution
In this case, the evolution of contributions would be like  this (`contributions` is an array of user contributions)
- contribute(0.5) - `contributions[0] = 0.5` - the default zero-index contribution gets assigned  
- contributeAndCombine(0.2) - `contributions[0] = 0.5 + 0.2 = 0.7`
- contributeAndCombine(0.4) - `contributions[0] = 0.7 + 0.4 = 1.1`
- contribute(0.1) - `contributions[0] = 1.1, contributions[1] = 0.1`
- contribute(0.25) - `contributions[0] = 1.1, contributions[1] = 0.1, contributions[2] = 0.25`
- contributeAndCombine(0.6) - `contributions[0] = 1.1 + 0.6 = 1.7, contributions[1] = 0.1, contributions[2] = 0.25` 
- contribute(1) - `contributions[0] = 1.7, contributions[1] = 0.1, contributions[2] = 0.25, contributions[3] = 1.0`

### something else?

Reply:
Seems like we have this resolved, right?  For now, it just always adds to your initial contributions.